formats/json-compact-refract.md

```diff
@@ -6,76 +6,75 @@ serialization of Refract, as seen in the Refract specification. It also allows
-normal array and a Compact Element. Because of this, in situations where there
-is ambiguity, the array SHOULD be treated as a normal array and handled
-accordingly. Ambiguity arises when an array is found in areas such as
-`attributes` or `content` and requires out-of-band information how to handle
-the array.
-
-#### Members
+A Refract element MUST be serialised as the following data structure as a JSON
+array.
 
 - (string, required) - Name of the element
-- (object, required) - Meta attributes of the element instance.
-  - `id` - Unique Identifier, MUST be unique throughout the document
-  - `ref` (Element Pointer) - Pointer to referenced element or type
-  - `classes` (array[string]) - Array of classifications for given element
-  - `title` (string) - Human-readable title of element
-  - `description` (string) - Human-readable description of element
-  - `links` (array[Link Element]) - Meta links for a given element
-- (object, required) - Attributes of the element instance
+- (object, nullable, required) - Meta attributes of the element instance
+    - `id` - Unique Identifier, MUST be unique throughout the document
+    - `ref` (Ref Element) - Pointer to referenced element or type
+    - `classes` (array[String Element]) - Array of classifications for given element
+    - `title` (string) - Human-readable title of element
+    - `description` (string) - Human-readable description of element
+    - `links` (array[Link Element]) - Meta links for a given element
+- (object, nullable, required) - Attributes of the element instance
 - (enum, required) - Element content with any of the following types
-  - (null)
-  - (string)
-  - (number)
-  - (boolean)
-  - (array)
-  - (object)
-  - (Compact Element)
-  - (array[Compact Element])
+    - Element
+    - array[Element]
+    - string
+    - boolean
+    - number
+    - null
+    - Key Value Pair
 
-## Example
+## Key Value Pair (array)
 
-In the base specification, serialization looks something like this for JSON:
+A Key Value Pair MUST be serialised a JSON array with the following values:
 
-```json
-{
-  "element": "foo",
-  "content": "bar"
-}
-```
+- (string, fixed): pair
+- (Element, required) - Key
+- (Element, optional) - Value
 
-In Compact Refract, this same example looks like this:
+## Example
+
+If I have a String Element with the value `Hello` it would be serialised in
+JSON Compact Refract as:
 
 ```json
-["foo", {}, {}, "bar"]
+["string", null, null, "Hello"]
 ```
 
 In this example, meta attributes and attributes are both required, even if the
 objects are empty. This is because this format relies on the structure of the
 array in order to determine values.
 
+If I attach a title meta attribute, the example would then be serialised as
+follows:
+
+```json
+[
+  "string",
+  {
+    "title": ["string", null, null, "My Title"]
+  },
+  null,
+  "Hello"
+]
+```
+
+See the [examples directory](examples/) for further examples.
```

Comment:
I'm referencing the examples from https://github.com/refractproject/refract-spec/pull/66 so this PR should be merged alongside.